### PR TITLE
copyonconflict: improve behavior with deleted files

### LINKED ISSUE #6: copy-on-conflict fails when file is deleted
Using unison-2.48.3 in batch mode with the option 'copy on conflict', I obtain an error message when a file has been deleted in one repository and changed in another.

**Steps for reproducing the error**
1. Create a new file test.txt in 'local' folder.
2. Run unison, so that it copies this file to the 'remote' folder.
3. Delete the file test.txt in the 'remote' folder.
4. Edit the file test.txt in the 'local' folder (fill it with some content).
5. Run unison again, with the options

```
batch = true
prefer = newer
copyonconflict = true
```

I get the following error message:

```
UNISON 2.48.3 started propagating changes at 21:41:01.66 on 01 Apr 2016
[BGN] Copying test.txt from [[/home/localfolder]] to [[//server//home/remotefolder]]
100%  00:00 ETAUnison failed: Uncaught exception File "/build/unison-HIcI1m/unison-2.48.3/copy.ml", line 1007, characters 17-23: Assertion failed
Raised at file "/build/unison-HIcI1m/unison-2.48.3/copy.ml", line 1007, characters 17-29
Called from file "/build/unison-HIcI1m/unison-2.48.3/copy.ml", line 1010, characters 2-22
Called from file "/build/unison-HIcI1m/unison-2.48.3/files.ml", line 87, characters 4-55
Called from file "/build/unison-HIcI1m/unison-2.48.3/files.ml", line 311, characters 17-57
Called from file "/build/unison-HIcI1m/unison-2.48.3/remote.ml", line 764, characters 4-24
Called from file "/build/unison-HIcI1m/unison-2.48.3/lwt/lwt.ml", line 75, characters 20-23
Re-raised at file "/build/unison-HIcI1m/unison-2.48.3/lwt/lwt.ml", line 135, characters 12-13
Called from file "list.ml", line 73, characters 12-15
Called from file "/build/unison-HIcI1m/unison-2.48.3/lwt/lwt.ml", line 31, characters 2-37
Called from file "/build/unison-HIcI1m/unison-2.48.3/lwt/lwt.ml", line 83, characters 17-46
Called from file "/build/unison-HIcI1m/unison-2.48.3/lwt/generic/lwt_unix_impl.ml", line 147, characters 6-40
Called from file "/build/unison-HIcI1m/unison-2.48.3/main.ml", line 202, characters 6-24
Called from file "/build/unison-HIcI1m/unison-2.48.3/main.ml", line 131, characters 4-9
```

[Client and server both run 64-bit versions of Ubuntu 14.04 and use identical builds of unison-2.48.3.]

### PR DESCRIPTION
**Pull request by tleedjarv**
Closes #6 (the issue itself has been fixed a long time ago, this PR improves the solution a bit).

Instead of silently ignoring deletion in one replica, create a conflict copy for the deleted file when the file was modified in one replica and deleted in the other replica.

### CODE REVIEW

**Review by gdt — APPROVED** [review]
I have no objections but would appreciate review from someone who understands this area of code.   Will merge >=18Z Sunday, earlier if reviews convince me.

**Comment by tleedjarv** [comment]
It would be great if someone who uses 'copyonconflict' could have a look and test it to see if it matches expectations.

**Pull request merged**
